spec/reservation_mgr_spec.rb

```diff
@@ -0,0 +1,157 @@
+    end
+  end
+
+  describe "available_rooms method" do
+    before do
+      @new_ResMgr = ReservationMgr.new(3)
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-10"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-30"),Date.parse("2018-10-07"))
+    end
+    it "returns array of available rooms" do
+      expect(@new_ResMgr.available_rooms(Date.parse("2018-09-05"),Date.parse("2018-09-07"))).must_be_kind_of Array
+      expect(@new_ResMgr.available_rooms(Date.parse("2018-09-05"),Date.parse("2018-09-07")).length).must_equal 1
+      expect(@new_ResMgr.available_rooms(Date.parse("2018-11-05"),Date.parse("2018-11-07")).length).must_equal 3
+    end
+    it "will raise an argument error if check_in date is greater than check_out date" do
+      expect{@new_ResMgr.available_rooms(Date.parse("2018-09-07"),Date.parse("2018-09-05"),1)}.must_raise ArgumentError
+    end
+    it "will raise an argument error if rooms requested is greater than rooms available" do
+      expect{@new_ResMgr.available_rooms(Date.parse("2018-09-07"),Date.parse("2018-09-05"),2)}.must_raise ArgumentError
+    end
+  end
+
+  describe "reserve_block method" do
+    before do
+      @new_ResMgr = ReservationMgr.new(6)
+    end
+    it "method reserves 5 rooms with false id" do
+      block = @new_ResMgr.reserve_block(Date.parse("2018-09-05"),Date.parse("2018-09-07"),5,"Metzner")
+      expect(block.length).must_equal 5
+      expect(@new_ResMgr.reservations[0].block_available).must_equal true
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect(@new_ResMgr.reservations.last.block_available).must_equal false
+    end
+    it "will raise an argument error if they try to block more than 5 rooms" do
+      expect{@new_ResMgr.reserve_block(Date.parse("2018-09-05"),Date.parse("2018-09-07"),6,"Metzner")}.must_raise ArgumentError
+    end
+    it "will raise an arguemnt error if there are not enough available rooms for a given date range" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect{@new_ResMgr.reserve_block(Date.parse("2018-09-05"),Date.parse("2018-09-07"),4,"Metzner")}.must_raise ArgumentError
+    end
+  end
+
+  describe "find_available_block_rooms method" do
+    before do
+      @new_ResMgr = ReservationMgr.new(6)
+    end
+    it "returns rooms in block that have not been reserved" do
+      block = @new_ResMgr.reserve_block(Date.parse("2018-09-05"),Date.parse("2018-09-07"),5,"Metzner")
+      expect(@new_ResMgr.find_available_block_rooms("Metzner").length).must_equal 5
+      expect(@new_ResMgr.find_available_block_rooms("Metzner")[0]).must_be_kind_of Reservation
```

Comment:
I would also check that the available block rooms doesn't include the one just reserved.